src/dbinterface.jl

```diff
@@ -384,11 +384,21 @@ Base.IteratorEltype(::Type{Cursor{false, knownlength}}) where {knownlength} = Ba
 Base.eltype(x::Cursor{false, knownlength}) where {knownlength} = Row
 
 function Base.iterate(x::Cursor{false}, st=1)
-    status = API.SQLFetchScroll(API.getptr(x.stmt), API.SQL_FETCH_NEXT, 0)
-    while status == API.SQL_STILL_EXECUTING
+    status = API.SQL_SUCCESS
+    while true
         status = API.SQLFetchScroll(API.getptr(x.stmt), API.SQL_FETCH_NEXT, 0)
+        status == API.SQL_STILL_EXECUTING && continue  # makes the usual case fast
+        if status == API.SQL_NO_DATA || status == API.SQL_NTS
```

Comment:
@quinnj returning nothing on `SQL_NTS` is based on the old behaviour, not sure if there is a bug hidden here. (previously it was implicit, if not a success code)